code/modules/mining/satchel_ore_boxdm.dm

```diff
@@ -53,8 +53,9 @@
 	user << browse(dat, "window=orebox")
 
 /obj/structure/ore_box/proc/dump_box_contents()
-	for(var/obj/item/ore/O in contents)
-		O.forceMove(loc)
+	for(var/obj/item/ore/O in src)
+		O.forceMove(drop_location())

```

Comment:
if you're going to sleep in the loop with tick check, you should sanity check the box and each item still exist on each iteration of the loop with ! QDELETED(thing) macro

Reply:
you can do this too:

```
var/drop = drop_location()
for(var/obj/item/ore/O in src)
    O.forceMove(drop)
    if(TICK_CHECK)
        stoplag()
        drop = drop_location()
```

no need to call ```drop_location()``` each iteration, only when it could feasibly have changed, ie: when the proc sleeps